drivers/sensor/bosch/bmm350/bmm350.c

```diff
@@ -14,74 +14,65 @@
 
 LOG_MODULE_REGISTER(BMM350, CONFIG_SENSOR_LOG_LEVEL);
 
-static inline int bmm350_bus_check(const struct device *dev)
+static int bmm350_read_otp_word(const struct device *dev, uint8_t addr, uint16_t *lsb_msb)
 {
-	const struct bmm350_config *cfg = dev->config;
-
-	return cfg->bus_io->check(&cfg->bus);
-}
-
-static inline int bmm350_reg_read(const struct device *dev, uint8_t start, uint8_t *buf, int size)
-{
-	const struct bmm350_config *cfg = dev->config;
-
-	return cfg->bus_io->read(&cfg->bus, start, buf, size);
-}
-
-int bmm350_reg_write(const struct device *dev, uint8_t reg, uint8_t val)
-{
-	const struct bmm350_config *cfg = dev->config;
-
-	return cfg->bus_io->write(&cfg->bus, reg, val);
-}
-
-static int8_t bmm350_read_otp_word(const struct device *dev, uint8_t addr, uint16_t *lsb_msb)
-{
-	int8_t ret = 0;
+	int ret;
 	uint8_t tx_buf = 0;
 	uint8_t rx_buf[3] = {0x00};
 	uint8_t otp_status = 0;
 	uint8_t otp_err = BMM350_OTP_STATUS_NO_ERROR;
 	uint8_t lsb = 0, msb = 0;
 
-	if (lsb_msb) {
-		/* Set OTP command at specified address */
-		tx_buf = BMM350_OTP_CMD_DIR_READ | (addr & BMM350_OTP_WORD_ADDR_MSK);
-		ret = bmm350_reg_write(dev, BMM350_REG_OTP_CMD_REG, tx_buf);
-		if (ret) {
-			LOG_ERR("i2c xfer failed! read addr = 0x%02x, ret = %d\n", tx_buf, ret);
-			return ret;
-		}
+	__ASSERT_NO_MSG(lsb_msb != NULL);
 
-		do {
-			/* Get OTP status */
-			ret += bmm350_reg_read(dev, BMM350_REG_OTP_STATUS_REG, &rx_buf[0], 3);
-			otp_status = rx_buf[2];
-			otp_err = BMM350_OTP_STATUS_ERROR(otp_status);
-			if (otp_err != BMM350_OTP_STATUS_NO_ERROR) {
-				break;
-			}
-		} while ((!(otp_status & BMM350_OTP_STATUS_CMD_DONE)) && (ret == BMM350_OK));
+	/* Set OTP command at specified address */
+	tx_buf = BMM350_OTP_CMD_DIR_READ | (addr & BMM350_OTP_WORD_ADDR_MSK);
+	ret = bmm350_reg_write(dev, BMM350_REG_OTP_CMD_REG, tx_buf);
+	if (ret != 0) {
+		LOG_ERR("i2c xfer failed! read addr = 0x%02x, ret = %d\n", tx_buf, ret);
```

Comment:
please consider dropping the "\n"s from all log statement, they're not necessary

Reply:
it looks like they were all there beforehand... but you're right they all should be removed